### PR TITLE
Feature/apply yaml

### PR DESCRIPTION
**Pull request by ChathurangaKCD**
image

### CODE REVIEW

File: src/modules/gitops/api.ts [hunk 1]
```diff
@@ -1,32 +1,53 @@
...
 export async function applyManifest(
   orgId: string,
   projectId: string,
   envId: string,
-  manifest: string
+  manifest: K8sObject
 ) {
-  const url = `${getAPIGateway()}/${endpoints.RUDDER_MAINFEST_IMPORT}`;
+  const url = `${getAPIGateway()}/${apiEndpoints.RUDDER_MAINFEST_IMPORT}`;
+  // const url = `http://localhost:3000/api/v1/import/manifest`;
```
Comment (binura-g) [inline src/modules/gitops/api.ts:15]: remove comment

File: src/modules/gitops/manifest-file.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  readonly manifest: K8sObject;
+  readonly subject: BehaviorSubject<ManifestState>;
+  matchedYamlObjects: YamlObject[] = [];
+  constructor(manifest: K8sObject) {
+    this.manifest = manifest;
+    this.subject = new BehaviorSubject<ManifestState>(ManifestState.WAITING);
+    skippedStateNotifier.subscribe(() => {
+      this.skipOnError();
+    });
+    // this.subject.subscribe({
```
Comment (binura-g) [inline src/modules/gitops/manifest-file.ts:99]: remove

**Review by binura-g — CHANGES_REQUESTED** [review]
Looks good @ChathurangaKCD. Just a few rogue comments that need to be removed but that's all.

On a separate note, how many different yamls have you tested this with?

File: src/commands/apply.ts [hunk 3]
```diff
@@ -54,18 +74,156 @@ export default class Apply extends Command {
...
+      cli.log('Manifests found: ', chalk.blue(manifestCount));
+      cli.log('Success count: ', chalk.blue(successCount));
+      cli.log(
+        'Errors occurred: ',
+        errCount === 0 ? chalk.blue(0) : chalk.red(errCount)
+      );
+      cli.log(
+        'Skipped manfests: ',
+        skippedCount === 0 ? chalk.blue(0) : chalk.yellow(skippedCount)
+      );
```
Comment (binura-g) [inline src/commands/apply.ts:174]: I think we should also print out the applied filenames and their statuses (especially if they fail/get skipped).

kubectl prints something like this when you apply a bunch of resources -
Deployment x created
ConfigMap y configured
....

And the errors as well, if it's actually human-readable. (idk what rudder actually returns on error)

**Review by binura-g — CHANGES_REQUESTED** [review]

File: src/commands/apply.ts (hunk 3 above)
Comment (ChathurangaKCD) [reply src/commands/apply.ts]: done
image

**Comment by ChathurangaKCD** [comment]
> On a separate note, how many different yamls have you tested this with?

@binura-g 30-40 yamls from [https://github.com/GoogleCloudPlatform/kubernetes-engine-samples](url)

**Review by binura-g — APPROVED** [review]
Looks good.

I've updated a few setup scripts for the CI. 

Can you run `yarn test` and fix the lining errors? (or turn them off in .eslintrc if they're useless). If you want to merge this PR and open another for it, that's fine as well.

**Comment by ChathurangaKCD** [comment]
will open another PR for linting errors

**Pull request merged**